src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -380,6 +446,113 @@ class _Call21Bool(Protocol):
         **kwds: Unpack[_Kwargs3],
     ) -> np.bool | NDArray[np.bool]: ...
 
+@type_check_only
+class _Call21Logical(Protocol):
+    @overload  # (scalar, scalar, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (scalar, scalar) -> bool
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.bool: ...
+    @overload  # (array-like, array, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
```

Comment:
```suggestion
        x1: _ArrayLikeNumber_co | _ArrayLikeObject_co,
```

where `_ArrayLikeObject_co` (in `numpy._typing`) is simply an alias for `_ArrayLike[np.object_]`